### PR TITLE
Add a type for logical literals

### PR DESCRIPTION
**Pull request by JordanBoltonMN**

### CODE REVIEW

**Comment by mattmasson** [comment]
Wait, what? What did we support before?

File: src/test/libraryTest/type/typeUtils/typeUtils.ts [hunk 1]
```diff
@@ -82,6 +82,26 @@ describe(`TypeUtils`, () => {
...
+            const expected: ReadonlyArray<AbridgedType> = [Type.LogicalInstance];
+            expect(actual).deep.equal(expected);
+        });
+
+        it(`prefer nullable in boolean simplification`, () => {
+            const actual: ReadonlyArray<AbridgedType> = TypeUtils.simplify([
+                TypeUtils.logicalLiteralFactory(false, "true"),
+                TypeUtils.logicalLiteralFactory(false, "false"),
+                TypeUtils.logicalLiteralFactory(true, "true"),
+                TypeUtils.logicalLiteralFactory(true, "false"),
```
Comment (mattmasson) [inline src/test/libraryTest/type/typeUtils/typeUtils.ts:99]: I don't understand this line.

**Comment by JordanBoltonMN** [comment]
> Wait, what? What did we support before?

We have always supported the type `logical`. This change support for logical literals, such as `true` and `false`, into the type system as child types of `logical`. Previously if you did a hover over `let foo = true in foo` it would say the type is `boolean`. Now it'll say `true`.

File: src/test/libraryTest/type/typeUtils/typeUtils.ts (hunk 1 above)
Comment (JordanBoltonMN) [reply src/test/libraryTest/type/typeUtils/typeUtils.ts:99]: It's creating the 4 possible variants of a logical literal (flag for true/flase, and a flag for isNullable). The test is to see if `simplify(true | nullable false)` resolves to `nullable boolean`.

Comment (JordanBoltonMN) [reply src/test/libraryTest/type/typeUtils/typeUtils.ts]: Simplified lines by creating TrueInstance/FalseInstance/etc.

**Review by mattmasson — APPROVED** [review]

**Pull request merged**